subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -1187,6 +1217,9 @@ void ull_scan_aux_release(memq_link_t *link, struct node_rx_pdu *rx)
 			struct lll_sync *sync_lll;
 			struct ll_sync_set *sync;
 
+			/* This code path is valid for Periodic Sync only */
+			LL_ASSERT(IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC));
+
```

Comment:
Shouldn't the `if (lll_aux) {` be `if (IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC) && lll_aux) {`

then? Seems weird to have an assert with this

Reply:
`lll_aux`  can have its `parent` as the scan context or sync context which is at runtime when receiving chains. If `scan` is NULL, and Periodic Sync feature is not supported in the build, I want it to assert.  

i can do this below, it that better?

```
} else if (IS__ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC)) {
    ...
} else {
    /* When not supporting Periodic Sync, the parent shall be scan context */
    LL_ASSERT(0);
    return;
}
```